File: 이것이 취업을 위한 코딩 테스트다/이진_탐색/부품_찾기/부품_찾기_set.cpp
#include <iostream>
#include <set>
using namespace std;

int main() {
  int n, m;
  set<int> arr;

  cin >> n;
  
  for (int i = 0; i < n; i++) {
    int num;
    cin >> num;

    arr.insert(num);
  }

  cin >> m;

  for (int i = 0; i < m; i++) {
    int num;
    cin >> num;

    if (arr.find(num) != arr.end()) {
      cout << "yes" << ' ';
    } else {
      cout << "no" << ' ';
    }
  }

  return 0;
}

File: 이것이 취업을 위한 코딩 테스트다/이진_탐색/부품_찾기/부품_찾기_set.py
n = int(input())
array = set(map(int, input().split()))
m = int(input())
customer = list(map(int, input().split()))

print(array)

for num in customer:
  if num in array:
    print('yes', end = ' ')
  else:
    print('no', end = ' ')
